### ISSUE TITLE
Dead kernel when fitting XGBoost model

### ISSUE THREAD

**Issue by avaserin** [dmlc/xgboost#9798]
I have been running XGBoost in Jupyter Notebooks using Anaconda successfully for months. However, I recently updated conda to 23.7.4 and XGBoost to 1.7.3, and, now when I fit an XGBoost model, a popup comes up that says "Kernel Restarting: The kernel appears to have died. It will restart automatically." I am running a very small dataframe (around 100 features and 80 rows), so I think it is an issue with my version updates. I am running all of the following versions for other packages for reference:

XGBoost version: 1.7.3
NumPy version: 1.24.3
SciPy version: 1.11.3
Pandas version: 2.1.1
Python version: 3.9.12 (main, Apr  5 2022, 01:53:17) 
[Clang 12.0.0 ]

Does anyone know how I can fix this issue?

**Comment by trivialfis** [comment]
Could you please:
- Update XGBoost to the latest (2.0.2) or (1.7.6)?
- Share your notebook if the update doesn't fix the issue?

**Comment by avaserin** [comment]
When I ran 

$ conda install xgboost=1.7.6 

in my terminal I got:

Collecting package metadata (current_repodata.json): done
Solving environment: unsuccessful initial attempt using frozen solve. Retrying with flexible solve.
Collecting package metadata (repodata.json): done
Solving environment: unsuccessful initial attempt using frozen solve. Retrying with flexible solve.

PackagesNotFoundError: The following packages are not available from current channels:

  - xgboost=1.7.6

Current channels:

  - https://repo.anaconda.com/pkgs/main/osx-64
  - https://repo.anaconda.com/pkgs/main/noarch
  - https://repo.anaconda.com/pkgs/r/osx-64
  - https://repo.anaconda.com/pkgs/r/noarch

To search for alternate channels that may provide the conda package you're
looking for, navigate to

    https://anaconda.org

and use the search bar at the top of the page.

**Comment by trivialfis** [comment]
Consider using `conda-forge` channel, https://anaconda.org/conda-forge/xgboost

**Comment by MCRE-BE** [comment]
Same issue with 1.7.6
2.0.2 is available nowhere on conda-forge.

**Comment by trivialfis** [comment]
Could you please share your notebook? Maybe in private?

**Comment by MCRE-BE** [comment]
What are you looking for? For me it crashes as part of a package. So the notebook calls classes that are hidden. Basically the notebook just calls : 
````
loop = Loop(**params)
loop.run_forecast()
````

**Comment by trivialfis** [comment]
I don't know why it crashes, unless I can debug the code myself I don't think I can find out the cause.

**Comment by trivialfis** [comment]
Basically,I need a https://en.m.wikipedia.org/wiki/Minimal_reproducible_example

**Comment by MCRE-BE** [comment]
Ok. Strangely, I don't think the problem is related to the code or the data. It's probably rather an install issue or an interaction with another package.

- So I thought 1.7.6 solved my issue as my code was running on Saturday and the kernel was not crashing. However, I got the error `module 'xgboost' has no attribute 'XGBRegressor`
- I tried reinstalling through mamba, which failed. 
- Reinstalled with conda 2 times and that just created an empty folder in my environment. I don't know if it comes from the conda install or through some interaction with pip (due to my pyproject file and requirements that were not 100% in line with my environment).

It seems to be solved now by deleting and reinstalling from scratch.
